src/transformers/trainer.py

```diff
@@ -4002,15 +4026,38 @@ def prediction_loop(
                 )
             self.control = self.callback_handler.on_prediction_step(args, self.state, self.control)
 
-            # Gather all tensors and put them back on the CPU if we have done enough accumulation steps.
-            if args.eval_accumulation_steps is not None and (step + 1) % args.eval_accumulation_steps == 0:
+            if self.args.batch_eval_metrics:
+                if self.compute_metrics is not None and preds_host is not None and labels_host is not None:
+                    is_last_step = step == len(dataloader) - 1
+                    if args.include_inputs_for_metrics:
+                        metrics = self.compute_metrics(
+                            EvalPrediction(predictions=preds_host, label_ids=labels_host, inputs=inputs_host),
+                            compute_result=is_last_step,
+                        )
+                    else:
+                        metrics = self.compute_metrics(
+                            EvalPrediction(predictions=preds_host, label_ids=labels_host),
+                            compute_result=is_last_step,
+                        )
+                del losses_host, preds_host, inputs_host, labels_host
+                torch.cuda.empty_cache()
+                losses_host, preds_host, inputs_host, labels_host = None, None, None, None
+
+            elif (
+                args.eval_accumulation_steps is not None
+                and (step + 1) % args.eval_accumulation_steps == 0
+                and self.accelerator.sync_gradients
+            ):
```

Comment:
It looks like the condition here was outdated, I updated it to the current version in `main`
```suggestion
            elif args.eval_accumulation_steps is not None and (step + 1) % args.eval_accumulation_steps == 0:
```